modules/packages/Python.chpl

```diff
@@ -1049,20 +1054,74 @@ module Python {
     /*
       Converts an array to a python set
     */
-    // proc toSet(arr): PyObjectPtr throws {
-    // TODO
-    // }
+    proc toSet(arr): PyObjectPtr throws
+      where isArrayType(arr.type) &&
+            arr.rank == 1 && arr.isDefaultRectangular() {
+      var pySet = PySet_New(nil);
+      this.checkException();
+      for i in 0..<arr.size {
+        const idx = arr.domain.orderToIndex(i);
+        PySet_Add(pySet, toPython(arr[idx]));
+        this.checkException();
+      }
+      return pySet;
+    }
 
-    // TODO: convert chapel set to python set
+    /*
+      Convert a chapel set to a python set. Returns a new reference
+    */
+    @chpldoc.nodoc
+    proc toSet(s: Set.set(?)): PyObjectPtr throws {
+      var pySet = PySet_New(nil);
+      this.checkException();
+      for i in s {
+        PySet_Add(pySet, toPython(i));
+        this.checkException();
+      }
+      return pySet;
+    }
 
     /*
       Converts a python set to an array
     */
-    // proc fromSet(type T, obj: PyObjectPtr): T throws {
-    // TODO
-    // }
+    proc fromSet(type T, obj: PyObjectPtr): T throws
+      where isArrayType(T) {
+
+      if (PySequence_Check(obj) == 0) then
+        throw new ChapelException("Can only convert a sequence with a known size to an array");
+
+      // if it's a sequence with a known size, we can just iterate over it
+      var res: T;
+      if PySet_Size(obj) != res.size.safeCast(Py_ssize_t) {
+        throw new ChapelException("Size mismatch");
+      }
```

Comment:
You first check if its a sequence, then call Pyset_size. Instead, consider using `PySet_Check`